From 21205200c1ab2a9c99d6d0419b1ce6623a3e053c Mon Sep 17 00:00:00 2001
From: Graham Percival <gperciva@tarsnap.com>
Date: Sun, 27 Sep 2015 17:18:58 -0700
Subject: [PATCH] Show command synopsis in usage()
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
Organization: Invisible Things Lab
Cc: Marek Marczykowski-Górecki <marmarek@invisiblethingslab.com>

Signed-off-by: Marek Marczykowski-Górecki <marmarek@invisiblethingslab.com>
---
 main.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/main.c b/main.c
index 091731a..42bd00a 100644
--- a/main.c
+++ b/main.c
@@ -42,7 +42,9 @@ usage(void)
 {
 
 	fprintf(stderr,
-	    "usage: scrypt {enc | dec} [...] infile [outfile]\n");
+	    "usage: scrypt {enc | dec} [-M maxmem] [-m maxmemfrac]"
+	    " [-t maxtime] infile\n"
+	    "              [outfile]\n");
 	exit(1);
 }
 
-- 
2.5.5

