From 2e4126a8608bc32ca70bfb5ef463ce73786c2e49 Mon Sep 17 00:00:00 2001
From: Graham Percival <gperciva@tarsnap.com>
Date: Wed, 27 Jan 2016 13:43:18 -0800
Subject: [PATCH] Add scrypt --version
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

---
 main.c   | 6 +++++-
 scrypt.1 | 4 ++++
 2 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/main.c b/main.c
index 5dcedf6..e20cdd9 100644
--- a/main.c
+++ b/main.c
@@ -45,7 +45,8 @@ usage(void)
 	fprintf(stderr,
 	    "usage: scrypt {enc | dec} [-M maxmem] [-m maxmemfrac]"
 	    " [-t maxtime] [-v] infile\n"
-	    "              [outfile]\n");
+	    "              [outfile]\n"
+	    "       scrypt --version\n");
 	exit(1);
 }
 
@@ -74,6 +75,9 @@ main(int argc, char *argv[])
 		maxtime = 5.0;
 	} else if (strcmp(argv[1], "dec") == 0) {
 		dec = 1;
+	} else if (strcmp(argv[1], "--version") == 0) {
+		fprintf(stdout, "scrypt %s\n", PACKAGE_VERSION);
+		exit(0);
 	} else
 		usage();
 	argc--;
diff --git a/scrypt.1 b/scrypt.1
index cdb05c9..1098e0c 100644
--- a/scrypt.1
+++ b/scrypt.1
@@ -35,6 +35,8 @@
 .Op Fl t Ar maxtime
 .Ar infile
 .Op Ar outfile
+.Nm
+.Fl -version
 .Sh DESCRIPTION
 .Nm Cm enc
 encrypts
@@ -67,6 +69,8 @@ of the available RAM to compute the derived encryption key.
 Use at most
 .Ar maxtime
 seconds of CPU time to compute the derived encryption key.
+.It Fl -version
+Print version of scrypt, and exit.
 .El
 .Pp
 In
-- 
2.5.5

