From 7314d33f3b28321291f3f19597177cc929a0a418 Mon Sep 17 00:00:00 2001
From: Ian Jackson <ijackson@chiark.greenend.org.uk>
Date: Thu, 3 Mar 2016 00:08:03 +0000
Subject: [PATCH 2/3] Provide -P option to read password from stdin
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Ian Jackson <ijackson@chiark.greenend.org.uk>
---
 main.c   | 6 +++++-
 scrypt.1 | 6 ++++++
 2 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/main.c b/main.c
index e20cdd9..42cb99a 100644
--- a/main.c
+++ b/main.c
@@ -55,6 +55,7 @@ main(int argc, char *argv[])
 {
 	FILE * infile;
 	FILE * outfile;
+	int devtty = 1;
 	int dec = 0;
 	size_t maxmem = 0;
 	double maxmemfrac = 0.5;
@@ -98,6 +99,9 @@ main(int argc, char *argv[])
 		GETOPT_OPT("-v"):
 			verbose = 1;
 			break;
+		GETOPT_OPT("-P"):
+			devtty = 0;
+			break;
 		GETOPT_MISSING_ARG:
 			warn0("Missing argument to %s\n", ch);
 			/* FALLTHROUGH */
@@ -134,7 +138,7 @@ main(int argc, char *argv[])
 
 	/* Prompt for a password. */
 	if (readpass(&passwd, "Please enter passphrase",
-	    dec ? NULL : "Please confirm passphrase", 1))
+	    (dec || !devtty) ? NULL : "Please confirm passphrase", devtty))
 		exit(1);
 
 	/* Encrypt or decrypt. */
diff --git a/scrypt.1 b/scrypt.1
index e5145c4..4738ad0 100644
--- a/scrypt.1
+++ b/scrypt.1
@@ -33,6 +33,7 @@
 .Op Fl M Ar maxmem
 .Op Fl m Ar maxmemfrac
 .Op Fl t Ar maxtime
+.Op Fl P
 .Ar infile
 .Op Ar outfile
 .Nm
@@ -61,6 +62,8 @@ reads passphrases from its controlling terminal, or failing that,
 from stdin.  Prompts are only printed if this means that
 .Nm
 is reading passphrases from some terminal.
+(But, see
+.Sm Fl P Li .)
 .Sh OPTIONS
 .Bl -tag -width "-m maxmemfrac"
 .It Fl M Ar maxmem
@@ -75,6 +78,9 @@ of the available RAM to compute the derived encryption key.
 Use at most
 .Ar maxtime
 seconds of CPU time to compute the derived encryption key.
+.It Fl P
+Always read passphrase from stdin, and do so only once even
+when encrypting.
 .It Fl -version
 Print version of scrypt, and exit.
 .El
-- 
2.5.5

